### DISCUSSION TITLE
Getting slug from getServerSideProps and remediating hook calls

### DISCUSSION THREAD

**Discussion by MikeCarbone** [vercel/next.js#28994] [Help]
Using a dynamic route, I need to get the slug from my URL before using my hooks to fetch data, specifically, the useSWR hooks as recommended by Vercel.

Right now, these are what my pages look like within the dynamically routed folder:

    // This will trigger server side loading
    export async function getServerSideProps(context) {
      const slug  = context.params.community
      if (!slug) {
         return { notFound: true }
      }
      return {
        props: { slug } // will be passed to the page component as props 
      }
    }
    
    
    // Slug gets passed in from above function, used in data fetching
    export default function Network ({ slug }) {
      // SWR call to fetch data
      const { community, isLoading, error } = useCommunityAdmin({ slug })
      ...
    }

&#x200B;

Note how the *only* thing my getServerSideProps is doing is getting the slug for the page. When Next sees this, it's going to render my page from the server every time. To my understanding, this takes away many of the benefits of using React in the first place and will hinder performance. (Components that stay the same between pages (( like a sidebar )) now have to get rerendered when navigating.)

I understand that I can *also* use useRouter to get my slug, but then my data fetch will run without the required slug parameter, and return errors. I can conditionally add logic somewhere down the pipe that prevents the hook from running, but that causes a whole new problem with conditionally calling hooks.

Am I missing something? Anybody have a tip or suggestion for how to do this correctly?

**Comment by wilsonneto-swe** [comment]
What you need is the getStaticProps and not getServerSide props.
getStaticProps will execute only one time by each slug
getServerSideProps will execute at every request

A good approach is to do the data fetch in getStaticProps, then you will execute this only one time and the next requests will just get a static response generated at this first time. You can combine getStaticProps with revalidate (ISG) to expire this static cached data after some time.

I hope it helps you.
I wish you success in your projects
Best regards
